packages/crayons-core/src/components/radio-group/radio-group.tsx

```diff
@@ -102,32 +170,41 @@ export class RadioGroup {
     let hasChecked = false;
 
     // Walk the DOM in reverse order, since the last selected one wins!
-    for (const radio of radios) {
+    for (const [index, radio] of radios.entries()) {
       if (!hasChecked && radio.value === value) {
         // correct value for this radio
         // but this radio isn't checked yet
         // and we haven't found a checked yet
         hasChecked = true;
         radio.checked = true;
+        radio.setAttribute('tabindex', '0');
+        this.selectedIndex = index;
       } else {
         // this radio doesn't have the correct value
         // or the radio group has been already checked
+        radio.setAttribute('tabindex', '-1');
         radio.checked = false;
       }
     }
 
     // Reset value if
     if (!hasChecked) {
+      console.log('radios:', radios);
+      radios.length !==0 && radios[0].setAttribute('tabindex', '0');
+      this.selectedIndex = 0;
       this.value = undefined;
     }
   }
 
-  private getRadios() {
-    return Promise.all(
-      Array.from(this.host.querySelectorAll('fw-radio')).map((r) =>
-        r.componentOnReady()
-      )
+  private async getRadios() {
```

Comment:
Yes Manoj, the option to move fetching radios to a lifecycle method will make code simple. I have moved the radios fetching code within `connectedCallback` as this will be the first one that gets executed when component is inserted in DOM and corresponding changes where `getRadios` was called is also made